### PR TITLE
Completing forEachOf

### PR DESCRIPTION
**Pull request by aearly**
This is mostly @dominicbarnes 's work from #168.  The last missing piece was documentation, which I added.  I also made sure it merges with what is currently on master, and simplified the logic in one place.

### CODE REVIEW

**Comment by dominicbarnes** [comment]
Nice! Would be great to finally see this make it in there!

File: lib/async.js [hunk 1]
```diff
@@ -68,6 +68,14 @@
         return memo;
     };
 
+    var _forEachOf = function (object, iterator) {
+        for (key in object) {
```
Comment (Adezandee) [inline lib/async.js]: May i ask why you don't use Object.keys() ?

File: test/test-async.js [hunk 2]
```diff
@@ -2806,20 +2999,20 @@ exports['cargo bulk task'] = function (test) {
 };
 
 exports['cargo drain once'] = function (test) {
-   
+
```
Comment (beaugunderson) [inline test/test-async.js]: I'm personally a fan of more vertical whitespace but these don't match the rest of the file and they seem extraneous to the work done in this PR. :)

Comment (aearly) [inline test/test-async.js]: These are trailing whitespace, no lines deleted.  My editor automatically does that.

**Comment by beaugunderson** [comment]
Oh oops, I misread. Totally fine then. :)

File: lib/async.js (hunk 1 above)
Comment (aearly) [inline lib/async.js]: No reason, I kept the original implementation.  I was focused more on merge conflicts.   `_keys` would be faster here...

**Comment by mjhasbach** [comment]
+1

**Comment by WillFrew** [comment]
+1

**Comment by felipou** [comment]
+1 (when will github add issue voting?)

**Pull request merged**